enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/protocol/ClientNettyPipelineBuilder.java

```diff
@@ -35,6 +36,7 @@
     public ClientNettyPipelineBuilder addFraming()
     {
         add( "frame_encoder", new LengthFieldPrepender( 4 ) );
+        add( "frame_decoder", new LengthFieldBasedFrameDecoder( Integer.MAX_VALUE, 0, 4, 0, 4 ) );
```

Comment:
`4` should be a constant somewhere.

Do we really need both encoder and decoder on both the server and client?

Reply:
In catchup, yes. In raft no, but it is safe.